drivers/android/process.rs

```diff
@@ -712,14 +706,15 @@ impl Process {
             return Ok(());
         }
 
-        let death = {
-            let mut pinned = Pin::from(death.write(
-                // SAFETY: `init` is called below.
-                unsafe { NodeDeath::new(info.node_ref.node.clone(), self.clone(), cookie) },
-            ));
-            pinned.as_mut().init();
-            Arc::<NodeDeath>::from(pinned)
-        };
+        let death = Arc::<NodeDeath>::from(
+            death
+                .pin_init_now::<core::convert::Infallible>(NodeDeath::new(
+                    info.node_ref.node.clone(),
+                    self.clone(),
+                    cookie,
+                ))
+                .map_err(|(i, _)| i)?,
```

Comment:
It is `(E, UniqueArc<T>)` and `E` is `Infallible`.

the function called is `UniqueArc::<MaybeUninit<T>>::pin_init_now(self, impl PinInit<T, E>) -> Result<UniqueArc<T>, (E, Self)>` and if the initializer fails I wanted to give a way to still keep the allocated memory.

This is the only place this function is used, because I did not want to change the code. I thought that there might be a reason that the memory is first allocated and then some checks are run (lines 701-707). If we can refactor this code to use `UniqueArc::pin_init` then we could remove the `pin_init_now` function (which I would prefer).